src/App/adapter/lib/state.js

```diff
@@ -63,27 +62,3 @@ export function parseState(state, config) {
 
   return [include, where]
 }
-
-const squash = (list) => {
-  const pairs = list.flatMap((obj) => Object.entries(obj))
-  return pairs.reduce((acc, scope) => ({ ...acc, [scope[0]]: scope[1] }), {})
-}
-
-const groupByLabel = (list) =>
-  list.reduce(
-    (acc, scope) => ({
-      ...acc,
-      [scope.label]: [...(acc[scope.label] || []), scope],
-    }),
-    {},
-  )
-
-export function mergeState(inits, diffs) {
-  const useful = diffs.reduce(
-    (acc, scope) => [...acc, scope.label, scope.group],
-    [LABEL_FOR_CONFIG],
-  )
-  const usefulInits = inits.filter(({ label }) => useful.includes(label))
-  const byLabel = groupByLabel([...usefulInits, ...diffs])
-  return Object.entries(byLabel).flatMap(([, a]) => squash(a))
-}
```

Comment:
If this code was dealing with an edge case of some sort that I have missed, please do let me know and I'll add a test case for it.